mne/io/fiff/tests/test_raw.py

```diff
@@ -115,6 +115,33 @@ def test_copy_append():
     assert_equal(data.shape[1], 2 * raw._data.shape[1])
 
 
+@testing.requires_testing_data
+def test_append_channels():
+    """Test raw splitting / re-appending channel types
+    """
+    raw = Raw(fif_fname, preload=True)
+    raw_nopre = Raw(fif_fname, preload=False)
+    raw_eeg = raw.pick_types(meg=False, eeg=True, copy=True)
+    raw_meg = raw.pick_types(meg=True, eeg=False, copy=True)
+    raw_new = raw_meg.append_channels([raw_eeg])
+    nchans = len(raw_new.ch_names)
+
+    assert_true([ch in raw_new.ch_names for ch in raw.ch_names])
+    assert_true(raw_new[:][0].shape == raw[:nchans][0].shape)
```

Comment:
and then you don't even need this check because it's done essentially by the `assert_array_equal` below